client.go

```diff
@@ -2178,6 +2178,67 @@ func (c *ToznySDKV3) ShareSecretWithUsername(ctx context.Context, params ShareSe
 	return nil
 }
 
+type UnshareSecretInfo struct {
+	SecretName       string
+	SecretType       string
+	UsernameToRevoke string
+}
+
+// UnshareSecretFromUsername revokes read access to secrets of provided name & type for this specific user
+// Calling client must be the owner of the secret for this to succeed.
+func (c *ToznySDKV3) UnshareSecretFromUsername(ctx context.Context, params UnshareSecretInfo) error {
+	if params.UsernameToRevoke == "" {
+		return fmt.Errorf("UnshareSecretFromUsername: Username to revoke must be provided")
+	}
+	// find the clientID for the username
+	searchParams := identityClient.SearchRealmIdentitiesRequest{
+		RealmName:         c.CurrentIdentity.Realm,
+		IdentityUsernames: []string{params.UsernameToRevoke},
+	}
+	identities, err := c.E3dbIdentityClient.SearchRealmIdentities(ctx, searchParams)
+	if err != nil {
+		return err
+	}
+	// Username doesn't match an identity, so return an error
+	if len(identities.SearchedIdentitiesInformation) < 1 {
+		return fmt.Errorf("UnshareSecretFromUsername: no identity found within realm with username %s", params.UsernameToRevoke)
+	}
+	// Find the sharing group
+	revokeClientID := identities.SearchedIdentitiesInformation[0].ClientID
+	ownerClientID, err := uuid.Parse(c.StorageClient.ClientID)
+	if err != nil {
+		return fmt.Errorf("UnshareSecretFromUsername: Client ID must be a valid UUID, got %s", c.StorageClient.ClientID)
+	}
+	// return an error if user tries to unshare secret from self
+	if revokeClientID == ownerClientID {
+		return fmt.Errorf("UnshareSecretFromUsername: Cannot unshare secret from self")
+	}
+	groupName := fmt.Sprintf("tozny.secret.%s.%s.%s.%s.%s", c.CurrentIdentity.Realm, ownerClientID, revokeClientID, params.SecretName, params.SecretType)
```

Comment:
I'm sure this line of code exists elsewhere, think about extracting it out to a helper method so that that way if the name scheme changes in the future we only have to change that method in one place.